### PR TITLE
[#40] Stop including `cryptography`.

### LINKED ISSUE #40: Don't include `cryptography`.
On a second thought, if we don't include `cryptography`, then it can be installed and packaged with SFTPPlus as needed. Which would save us from re-releasing Pythia whenever there is a `cryptography` security update… Maybe useful after all?

_Originally posted by @dumol in https://github.com/chevah/pythia/issues/39#issuecomment-1503098629_

### PR DESCRIPTION
**Pull request by dumol**
Scope
-----

Fixes #40 

Changes
-------

Stop including the upstream `cryptography` wheels.

Building our own OpenSSL libs is still needed for Python's `ssl` module.  (Except on Windows, where the upstream package comes with its own OpenSSL libs.)

**Drive-by changes**:
  * updated Python to latest 3.11.x version
  * updated OpenSSL for Python's `ssl` to latest 3.1.x version on non-Windows platforms
  * minor lib updates to latest versions for XZ and SQLite
  * some other minor Python module updates, including a security fix for `pip`
  * minor workflow improvements from `server` repo.

Testing
-------

Check changes and automated tests.

Try the testing packages from bin.chevah.com, e.g. https://bin.chevah.com:20443/testing/3.11.7.3785e41/

### CODE REVIEW

**Review by adiroiban — APPROVED** [review]
Thanks. This is a good start.

PyNacl can be also removed... without problems.

With the removal of `cryptography` , things are getting ugly and complicated for generating the install version of chevah/server for Windows and MacOS or ... Linux ARM64

We are building the distributables only on Linux X64...
so it's complicated to install a Windows or MarcOS package on Linux.

But I think that for pythia we should keep things simple, and focus on Python

I have created a sepate ticket to deal with the chevah/server  https://github.com/chevah/server/issues/6480


Thanks

**Comment by adiroiban** [comment]
Also, happy to see Laja at work. I am quite happy with macmini M1 for macOS

GitHub now has mac M1 runners... but they cost $0.16 per minute... so with our 10 minutes job it would take more than $1.5 for each run

File: src/Python/chevahbs [hunk 1]
```diff
@@ -90,6 +90,11 @@ chevahbs_try() {
         # Locales not supported on Alpine 3.12, failing locale-related tests.
         (>&2 echo -e "\tNot running Python upstream tests on musl-based Linux.")
     else
+        if [ "$PYTHON_BUILD_VERSION" = "3.11.7" ]; then
```
[the remark this answers is not in the document]
Comment (dumol) [reply src/Python/chevahbs:93]: Thanks, this worked! And I've only disabled asyncio streams tests for the current Python version.

File: build.conf [hunk 2]
```diff
@@ -32,21 +32,18 @@ SQLITE_VERSION="3430100"
...
+PIP_VERSION="23.3.1"
+SETUPTOOLS_VERSION="69.0.2"
 
 # Python modules versions to be used everywhere possible.
 # Latest cryptography/bcrypt require Rust, use only wheels for them.
 CFFI_VERSION="1.16.0"
-# When updating cryptography, also update "expecting_openssl_version" in
-# the file src/chevah-python-tests/test_python_binary_dist.py to pass tests.
-CRYPTOGRAPHY_VERSION="41.0.4"
 PYNACL_VERSION="1.5.0"
```
[the remark this answers is not in the document]
Comment (dumol) [reply build.conf]: No problem, done trough https://github.com/chevah/pythia/pull/54/commits/844422bb97f7d704cefa99668bfa430ff31e0f99.

**Pull request merged**